lib/apiservers/service/restapi/handlers/common.go

```diff
@@ -152,3 +163,12 @@ func getVCHConfig(op trace.Operation, d *data.Data) (*config.VirtualContainerHos
 
 	return vchConfig, nil
 }
+
+func setAuth(b *BuildDataParams, p interface{}) {
+	if c, ok := p.(Credentials); ok {
+		b.user = c.user
+		b.password = c.pass
+	} else {
+		b.ticket = p.(Session).ticket
```

Comment:
Can we be defensive here and make this `} else if s, ok := p.(Session); ok {`?

That way we can have an `else` block so we don't panic if `p` isn't either.

Reply:
I think that's a case where you want to panic, as such a case would be a programmer error rather than error due to user input.

Reply:
Also, we shouldn't really get this far without one or the other as swagger will throw an auth error (though we probably shouldn't rely on that behavior).